addons/cookoff/functions/fnc_cookOffEffect.sqf

```diff
@@ -0,0 +1,187 @@
+#include "script_component.hpp"
+/*
+ * Author: Brandon (TCVM)
+ * Spawn cook-off effects
+ *
+ * Arguments:
+ * 0: Vehicle <Object>
+ * 1: Spawn fire jet <Boolean>
+ * 2: Spawn fire ring <Boolean>
+ * 3: How long effect will last (Max 20 seconds) <Number>
+ * 4: What selection will fire originate from <String>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [vehicle player, true, false, 15, "commander_turret"] call ace_cookoff_fnc_cookOffEffect
+ *
+ * Public: No
+ */
+
+params ["_obj", "_jet", "_ring", "_time", "_fireSelection"];
+private _light = "#lightpoint" createVehicleLocal [0,0,0];
+_light setLightBrightness 5;
+_light setLightAmbient [0.8, 0.6, 0.2];
+_light setLightColor [1, 0.5, 0.2];
+_light lightAttachObject [_obj, [0,0,0]];
+_time = 0 max (_time min 20);
+
+private _sound = objNull;
+if (isServer) then {
+    // ironically biggest performance hit is this. Creating a new sound source takes up aprox 400 milliseconds.
+    // I dont think there is an alternative that takes into effect distance and whatever, but if you find one please fix!
```

Comment:
Is it not faster to have a global event for the clients that players a sound locally, check the distanc and delay for speed of sound (and you can limit it to a radius etc)?

Reply:
A few reasons: 1) you can't stop sounds not soundsourced mid way iirc. It would weird to have the full 20 second audio play for a cooking that is 5 seconds long. The performance hit is substantial (300 ms profiled I think), but there is no better option without sacrificing features. 2) commy would block merging until his cba speed of sound audio feature merges and I would like this in ace sometime this year :P

Reply:
What about creating particle source and say3d from it? That also should allow you to stop the sound and can be done locally via global event.

Reply:
99% sure the CBA speed of sound framework is going to do that and was written for cookoff. I'd just forgotten about it...